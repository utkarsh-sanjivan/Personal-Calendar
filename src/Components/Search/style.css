.search-icon {
    cursor: pointer;
}

.autocomplete-input {

}