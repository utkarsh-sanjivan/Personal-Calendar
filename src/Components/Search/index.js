import React from 'react';
import { Input, Tooltip, AutoComplete } from 'antd';
import { SearchOutlined, CalendarOutlined } from '@ant-design/icons';
import './style.css';

const SearchField = props => (
        <AutoComplete 
            {...props}
            options={props.autoCompleteData}
        >
        <Input 
            placeholder="Search Events"
            prefix={
                <Tooltip title="Search Event">
                    <SearchOutlined className="search-icon"/>
                </Tooltip>
            }
            suffix={<CalendarOutlined />}
        />
        </AutoComplete>
)

export default SearchField;
