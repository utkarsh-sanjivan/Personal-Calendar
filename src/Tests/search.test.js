import React from 'react';
import { create } from "react-test-renderer";
import Search from './../Components/Search/index';

test('Test Search Component', () => {
const component = create(<Search 
    className=""
    onSearch={() =>{}}
    onSelect={() =>{}}
    autoCompleteData={[]}
  />);
  expect(component.toJSON()).toMatchSnapshot()
});