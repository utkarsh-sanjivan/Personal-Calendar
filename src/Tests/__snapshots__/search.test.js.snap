// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search Component 1`] = `
<div
  autoCompleteData={Array []}
  className="ant-select ant-select-auto-complete ant-select-single ant-select-customize-input ant-select-show-search"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="ant-select-selector"
    onMouseDown={[Function]}
  >
    <span
      className="ant-select-selection-search"
    >
      <span
        className="ant-select-selection-search-input ant-input-affix-wrapper"
        onMouseUp={[Function]}
        style={
          Object {
            "opacity": null,
          }
        }
      >
        <span
          className="ant-input-prefix"
        >
          <span
            aria-label="search"
            className="anticon anticon-search search-icon"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
        </span>
        <input
          aria-activedescendant="undefined_list_0"
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autoComplete="off"
          autoFocus={true}
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onPaste={[Function]}
          placeholder="Search Events"
          readOnly={false}
          role="combobox"
          style={null}
          type="text"
          unselectable={null}
          value=""
        />
        <span
          className="ant-input-suffix"
        >
          <span
            aria-label="calendar"
            className="anticon anticon-calendar"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="calendar"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
    <span
      className="ant-select-selection-placeholder"
    />
  </div>
</div>
`;
